wire-schema/src/main/java/com/squareup/wire/schema/IdentifierSet.kt

```diff
@@ -47,12 +47,31 @@ package com.squareup.wire.schema
 class IdentifierSet private constructor(builder: Builder) {
   private val includes = builder.includes.toSet()
   private val excludes = builder.excludes.toSet()
+  private val oldest = builder.oldest
+  private val newest = builder.newest
   private val usedIncludes = mutableSetOf<String>()
   private val usedExcludes = mutableSetOf<String>()
 
   val isEmpty: Boolean
     get() = includes.isEmpty() && excludes.isEmpty()
 
+  /** Returns true unless [options] specifies a version that is outside of the configured range. */
+  fun isRetainedVersion(options: Options): Boolean {
+    if (newest != null) {
+      val sinceOption = options[SINCE]
+      val since = (sinceOption as? String)?.toLongOrNull()
```

Comment:
```suggestion
      val since = (sinceOption as? String)?.toLong()
```
Shouldn't a specified but invalid format throw?

Reply:
Good call.